Orange/canvas/gui/utils.py

```diff
@@ -305,3 +305,18 @@ def message(icon, text, title=None, informative_text=None, details=None,
         mbox.setDefaultButton(default_button)
 
     return mbox.exec_()
+
+
+def OSX_NSURL_toLocalFile(url):
+    """Return OS X NSURL file reference as local file path or '' if not NSURL"""
+    if isinstance(url, QUrl):
+        url = url.toString()
+    if not url.startswith('file:///.file/id='):
+        return ''
+    from subprocess import Popen, PIPE, DEVNULL
+    # OS X Foundation, and with it NSURL, is available for default Python 2
+    return Popen('''python -c '
+import sys
+from Foundation import NSURL
```

Comment:
Ahh, seems this [only works if PyObjC package is installed](https://stackoverflow.com/questions/1614648/importerror-no-module-named-foundation) (requires building with Xcode).